chainstate/src/detail/tests/test_framework.rs

```diff
@@ -244,4 +238,9 @@ impl BlockTestFramework {
     pub fn chainstate(&mut self) -> &mut Chainstate {
         &mut self.chainstate
     }
+
+    pub fn index_at(&self, at: usize) -> &BlockIndex {
+        assert!(at > 0, "No block index for genesis");
+        &self.block_indexes[at - 1]
+    }
```

Comment:
Only "proper" block indices (not genesis) are stored in the `block_indexes` vector. That introduces an off-by-one array access issue, so it is better accessed through this function

Reply:
To be honest, the block_indexes vector shouldn't exist. The chainstate object should be used, specifically.

Reply:
Yeah, sure... we can look into this later. Don't worry about it.

Reply:
Unresolved, so it's still visible to others